packages/js/product-editor/src/hooks/use-validation.ts

```diff
@@ -0,0 +1,65 @@
+/**
+ * External dependencies
+ */
+import { useDispatch } from '@wordpress/data';
+import { useEffect, useState } from '@wordpress/element';
+
+/**
+ * Signals that product saving is locked.
+ *
+ * ```typescript
+ * // Sync validation
```

Comment:
Love these examples!  It might be worthwhile to move these to a MD file for discoverability and easier parsing once the docs project gets underway.